fission/src/test/scene/SceneRenderer.test.ts

```diff
@@ -0,0 +1,509 @@
+                update: vi.fn(),
+                setup: vi.fn(),
+                id: 0,
+            }
+            const mockSceneObject2: MockSceneObject = {
+                dispose: vi.fn(),
+                update: vi.fn(),
+                setup: vi.fn(),
+                id: 0,
+            }
+
+            sceneRenderer.registerSceneObject(mockSceneObject1 as unknown as SceneObject)
+            sceneRenderer.registerSceneObject(mockSceneObject2 as unknown as SceneObject)
+            expect(sceneRenderer.sceneObjects.size).toBe(2)
+
+            sceneRenderer.removeAllSceneObjects()
+            expect(sceneRenderer.sceneObjects.size).toBe(0)
+            expect(mockSceneObject1.dispose).toHaveBeenCalled()
+            expect(mockSceneObject2.dispose).toHaveBeenCalled()
+        })
+    })
+
+    describe("Geometry Creation", () => {
+        test("should create sphere with custom material", () => {
+            const customMaterial = new THREE.MeshBasicMaterial({ color: 0xff0000 })
+            const sphere = sceneRenderer.createSphere(1.0, customMaterial)
+            expect(sphere.material).toBe(customMaterial)
+        })
+
+        test("should create sphere with default material", () => {
+            const sphere = sceneRenderer.createSphere(1.0)
+            expect(sphere.material).toBeInstanceOf(THREE.MeshToonMaterial)
+        })
+
+        test("should create box with default material and correct position", () => {
+            // eslint-disable-next-line @typescript-eslint/naming-convention
+            const mockVec3: MockVec3 = { GetX: () => 2, GetY: () => 3, GetZ: () => 4 }
+            const box = sceneRenderer.createBox(mockVec3 as unknown as Parameters<typeof sceneRenderer.createBox>[0])
+            expect(box.material).toBeInstanceOf(THREE.MeshToonMaterial)
+            expect(box.geometry.attributes.position.array[0]).toBe(1)
+            expect(box.geometry.attributes.position.array[1]).toBe(1.5)
+            expect(box.geometry.attributes.position.array[2]).toBe(2)
+        })
+
+        test("should create toon material", () => {
+            const material = sceneRenderer.createToonMaterial(0xff0000, 3)
+            expect(material).toBeInstanceOf(THREE.MeshToonMaterial)
+            expect(material.color.getHex()).toBe(0xff0000)
+        })
+    })
+
+    describe("Coordinate Conversion", () => {
+        test("should convert center screen to world space", () => {
+            const centerX = window.innerWidth / 2
+            const centerY = window.innerHeight / 2
+            const worldPos = sceneRenderer.pixelToWorldSpace(centerX, centerY)
+
+            expect(worldPos.x).toBe(0)
+            expect(worldPos.y).toBe(0)
+        })
+
+        test("should convert world to pixel space", () => {
+            const pixelPos = sceneRenderer.worldToPixelSpace(new THREE.Vector3(0, 0, 0))
+            expect(pixelPos[0]).toBeCloseTo(1861, 0)
+            expect(pixelPos[1]).toBeCloseTo(1261, 0)
```

Comment:
Shouldn't these be based off of the window.innerWidth and window.innerHeight property like the previous test?